lib/controllers/skill-infrastructure-controller/index.js

```diff
@@ -140,12 +140,20 @@ module.exports = class SkillInfrastructureController {
      * @param {Function} callback (error)
      */
     updateSkillManifestWithDeployResult(rawDeployResult, callback) {
-        // 1.update local skill.json file: update the "uri" in all the existing "apis" for each region
-        R.keys(Manifest.getInstance().getApis()).forEach((domain) => {
+        const targetDomains = ResourcesConfig.getInstance().getTargetDomains(this.profile);
+        // for backward compatibility, defaulting to api from skill manifest if targetEndpoints is not defined
+        const domains = targetDomains.length ? targetDomains : Object.keys(Manifest.getInstance().getApis());
+        // 1.update local skill.json file: update the "uri" in all target endpoints for each region
+        domains.forEach((domain) => {
             R.keys(rawDeployResult).forEach((region) => {
                 Manifest.getInstance().setApisEndpointByDomainRegion(domain, region, rawDeployResult[region].endpoint);
             });
         });
+
+        if (ResourcesConfig.getInstance().hasEventsEndpoint(this.profile)) {
+            Manifest.getInstance().setEventEndpoint(rawDeployResult.default.endpoint.uri);
```

Comment:
what if the endpoint is not for default? the "events" object also has default endpoint uri and "regions" endpoint uri.

Reply:
updated